src/Illuminate/Http/Client/Batch.php

```diff
@@ -0,0 +1,389 @@
+
+        return $this;
+    }
+
+    /**
+     * Register a callback to run after a request from the batch fails.
+     *
+     * @param  (\Closure($this, int|string, \Illuminate\Http\Response|\Illuminate\Http\Client\RequestException): void)  $callback
+     * @return Batch
+     */
+    public function catch(Closure $callback): self
+    {
+        $this->catchCallback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Register a callback to run after all the requests from the batch succeed.
+     *
+     * @param  (\Closure($this, array<int|string, \Illuminate\Http\Response>): void)  $callback
+     * @return Batch
+     */
+    public function then(Closure $callback): self
+    {
+        $this->thenCallback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Register a callback to run after all the requests from the batch finish.
+     *
+     * @param  (\Closure($this, array<int|string, \Illuminate\Http\Response>): void)  $callback
+     * @return Batch
+     */
+    public function finally(Closure $callback): self
+    {
+        $this->finallyCallback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Send all of the requests in the batch.
+     *
+     * @return array<int|string, \Illuminate\Http\Response|\Illuminate\Http\Client\RequestException>
+     */
+    public function send(): array
+    {
+        $this->inProgress = true;
+
+        if ($this->beforeCallback !== null) {
+            call_user_func($this->beforeCallback, $this);
+        }
+
+        $results = [];
+        $promises = [];
+
+        foreach ($this->requests as $key => $item) {
+            $promise = match (true) {
+                $item instanceof PendingRequest => $item->getPromise(),
+                default => $item,
+            };
+
+            $promises[$key] = $promise;
+        }
+
+        if (! empty($promises)) {
+            (new EachPromise($promises, [
+                'fulfilled' => function ($result, $key) use (&$results) {
+                    $results[$key] = $result;
+
+                    $this->decrementPendingRequests();
+
+                    if ($result instanceof Response && $result->successful()) {
+                        if ($this->progressCallback !== null) {
+                            call_user_func($this->progressCallback, $this, $key, $result);
+                        }
+
+                        return $result;
+                    }
+
+                    if (($result instanceof Response && $result->failed()) ||
+                        $result instanceof RequestException) {
+                        $this->incrementFailedRequests();
+
+                        if ($this->catchCallback !== null) {
+                            call_user_func($this->catchCallback, $this, $key, $result);
+                        }
+                    }
+
+                    return $result;
+                },
+                'rejected' => function ($reason, $key) use ($catchCallback) {
```

Comment:
@WendellAdriel There is an alert on this line with the use ($catchCallback). Is this used? Looks like it's only using $this->catchCallback and the use statement could be deleted